passes/opt/opt_expr.cc

```diff
@@ -259,6 +259,20 @@ bool is_one_or_minus_one(const Const &value, bool is_signed, bool &is_negative)
 	return last_bit_one;
 }
 
+int get_highest_hot_index(RTLIL::SigSpec signal)
+{
+	for (int i = GetSize(signal) - 1; i >= 0; i--)
+	{
+		if (signal[i] == RTLIL::State::S0)
+			continue;
```

Comment:
Wouldn't the function do exactly the same thing without those two lines, since `(signal[i] == RTLIL::State::S0)` would imply `(signal[i] != RTLIL::State::S1)`.

Reply:
What about Sx?

Reply:
Sx would also imply `(signal[i] != RTLIL::State::S1)`, thus we'd reach the end of the loop body, which effectively is the same as `continue`. Same for non-constant signal bits.

Reply:
hm, you are right. What is the proper way to deal with Sx here anyhow? I think I tried to do something but failed.